scripts/companionmanager.js

```diff
@@ -136,6 +136,22 @@ class CompanionManager extends FormApplication {
     await this.wait(AECONSTS.animationFunctions[animation].time);
     //get custom data macro
     const customTokenData = await this.evaluateExpression(game.macros.getName(`AE_Companion_Macro(${actor.name})`)?.command, {summon: actor,spellLevel: this.spellLevel || 0, duplicates: duplicates, assignedActor: this.caster || game.user.character || _token.actor}) || {};
+    if (!customTokenData.actor) {
+      customTokenData.actor = {};
+    }
+    if (!customTokenData.actor.flags) {
+      customTokenData.actor.flags = {};
+    }
+    if (!customTokenData.token) {
+      customTokenData.token = {};
+    }
+    if (game.settings.get(AECONSTS.MN, "copyDisposition")) {
+      const summonerDisposition = (this.caster || game?.user?.character || _token?.actor)?.prototypeToken?.disposition;
+      if (typeof summonerDisposition !== "undefined") {
+        customTokenData.token.disposition = summonerDisposition;
+      }
+    }
+    customTokenData.actor.flags["automated-evocations"] = { "summonerID": (this.caster || game?.user?.character || _token?.actor)?.id || "", "spellLevel": this.spellLevel || 0 };
     customTokenData.elevation = posData?.flags?.levels?.elevation ?? _token?.document?.elevation ?? 0;
     customTokenData.elevation = parseFloat(customTokenData.elevation);

```

Comment:
I am wondering if this should actually be `customTokenData.token.elevation` instead of `customTokenData.elevation`, as Warpgate seems to expect subobjects for `actor` and `token` data in this param, and whether that would remove the need for the hooks in the lines below this?

Didn't touch it because I didn't want to break anything though :)